openlibrary/plugins/openlibrary/lists.py

```diff
@@ -1,20 +1,33 @@
 """Lists implementaion.
 """
 import random
+import six
 import web
 
 from infogami.utils import delegate
 from infogami.utils.view import render_template, public
 from infogami.infobase import client, common
 
-import six
-
-from openlibrary.core import formats, cache
+from openlibrary.core import cache
+from openlibrary.core import formats
+from openlibrary.core.lending import add_availability
 import openlibrary.core.helpers as h
 from openlibrary.utils import dateutil
 from openlibrary.plugins.worksearch import subjects
 
 
+@public
+def get_book_availability(lst):
```

Comment:
Lists are a mess.. by default they just return a list of `seeds` which have very limited metadata about the underlying objects (e.g. works, editions). Our canonical way to `add_availability` to books requires a list of works/editions. So, we use this function to create a lookup table containing works/editions from the list, keyed by their ocaid (so on the front-end, as we iterate over the list items, if they are in this table, we can easily get the underlying item/object and its availability (to pass to LoanStatus macro, to render CTA button)